op-e2e/e2eutils/setup.go

```diff
@@ -283,3 +284,9 @@ func ApplyDeployConfigForks(deployConfig *genesis.DeployConfig) {
 	// Activated by default, contracts depend on it
 	deployConfig.L1CancunTimeOffset = new(hexutil.Uint64)
 }
+
+func MainnetL1ChainConfigWithChainID(chainID *big.Int) *params.ChainConfig {
+	cfg := params.MainnetChainConfig
+	cfg.ChainID = chainID
+	return cfg
+}
```

Comment:
We likely don't want to have something like this, as the l1 chain config should be available when we are setting up the "real L1 config".